tests/accessor/local_accessor_access_among_work_items.h

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides tests that the sycl::local_accessor can access the memory shared
+//  among work-items.
+//
+*******************************************************************************/
+#ifndef SYCL_CTS_LOCAL_ACCESSOR_ACCESS_AMONG_WORK_ITEMS_H
+#define SYCL_CTS_LOCAL_ACCESSOR_ACCESS_AMONG_WORK_ITEMS_H
+#include "accessor_common.h"
+
+namespace local_accessor_access_among_work_items {
+using namespace sycl_cts;
+using namespace accessor_tests_common;
+
+/**
+ * @brief Provides a functor that provides verification that local_accessor can
+ *        access the memory shared among work-items
+ * @tparam T Current data type
+ * @tparam DimensionTypeT Current current dimension size
+ */
+template <typename T, typename DimensionTypeT>
+class run_test {
+  static constexpr int Dimension = DimensionT::value;
+
+ public:
+  /**
+   * @brief Functor that provides verification that local_accessor can access
+   *        the memory shared among work-items
+   * @param type_name Current data type string representation
+   */
+  void operator()(const std::string& type_name) {
+    auto queue = util::get_cts_object::queue();
+
+    auto section_name = get_section_name<Dimension>(
+        type_name,
+        "Verify possibility to access the memory shared among work-items. "
+        "[local_accessor]");
+    SECTION(section_name) {
+      T values_arr[2] = {1, 2};
+      const sycl::range range(2);
+      constexpr T valid_value = 5;
+      constexpr T invalid_value = 6;
+
+      bool is_acc_val_equal_to_expected = false;
+      {
+        sycl::buffer<bool> val_is_equal_to_expected_buffer(
+            &is_acc_val_equal_to_expected, sycl::range(1));
+        queue.submit([&](sycl::handler& cgh) {
+          auto val_is_equal_to_expected_acc =
+              val_is_equal_to_expected_buffer
+                  .template get_access<sycl::access_mode::write>(cgh);
+          sycl::local_accessor<T, Dimension> acc(range, cgh);
+          cgh.parallel_for(
+              sycl::nd_range(range, range), [=](sycl::nd_item item) {
+                auto lid = item.get_local_id(0);
+                // Initialize local memory with invalid value
+                acc[lid] = invalid_value;
+                // Wait for work-items to finish initialization
+                sycl::group_barrier(item.get_group());
+
+                // Work-items with index greater than 0 writes to valid data to
+                // the first element of the local data
+                if (lid != 0) {
+                  acc[0] = valid_value;
```

Comment:
It looks to me like a race condition.
Why not having each work-item writing its own id into `acc[(id + 1)%item.get_local_range(0)]` and 
checking later that `acc[(id + 7)%item.get_local_range(0)] == (id + 6)%item.get_local_range(0)` for example.

Reply:
Why is this a data race?
There are just 2 work-items, so we are definitely safe. Even with the range size > 2, the result of this kernel must be stable.

Reply:
Ah yes, good point I forgot there are only 2 work-items. :-)
If there are more than 2 work-items, where is it written in the memory model description that writing the same values in a data race is not a data race?
At least the test could be something like
```c++
acc[(id + 1)%item.get_local_range(0)] = id;
sycl::group_barrier(item.get_group()); 
global_acc[(id + 1)%item.get_local_range(0)] = (id + 2)%item.get_local_range(0);
```
and check on the host that the global buffer has the right iota values to avoid a compliant implementation with any local accessor implemented with a global buffer with just an element.

Reply:
Sorry, I agree it's a data race, but I don't think it's a problem. The execution order doesn't make the result undefined.

On the other hand I don't mind changing the code to avoid the data race.

Reply:
Anyway, all this might be some over-engineering an just the minimal test in this PR can be enough. As you want.